### DISCUSSION TITLE
transform nest import ?

### DISCUSSION THREAD

**Discussion by crapthings** [swc-project/swc#1800] [Ideas]
do we support such feature?

```js
function nestImport () {
  return import something from './something'
}
```

**Comment by kdy1** [comment]
No, it's not supported

**Reply by crapthings** [reply]
i try to replace meteor/babel to swc, but failed with nested import

https://github.com/meteor/meteor-feature-requests/issues/398

https://guide.meteor.com/1.4-migration.html#nested-imports

> Use Nested Imports
Thanks to the use of the reify library, Meteor now fully supports nested import declarations in both application and package modules, whereas previously they were only allowed in application code:

```js
if (Meteor.isClient) {
  import { symbol } from './client-only/file';
}
```

One place this is particularly useful is in test files that are only intended to run on the client or the server — you can now use import wherever you like, without having to organize your tests in client or server directories.

**Reply by afrokick** [reply]
@crapthings did you solve this issue? Or is there any workaround like plugin for swc to handle it?

**Reply by Austaras** [reply]
There doesn't seem to be possible to modify ast type of swc in the near future

**Reply by kdy1** [reply]
This will not be supported and not possible (even with plugins) as it requires wrong AST type definitions
